pubsub/google/cloud/pubsub_v1/subscriber/_protocol/streaming_pull_manager.py

```diff
@@ -292,13 +299,19 @@ def _maybe_release_messages(self):
             except queue.Empty:
                 break
```

Comment:
Also, do you not need to take a lock around

if self.load >= _MAX_LOAD:

above?

Reply:
Yes, a lock is indeed needed, although not in the `_maybe_release_messages()` method itself.

The method assumes that its caller has obtained the lock (mentioned in the docstring), which happens in [maybe_resume_consumer()](https://github.com/googleapis/google-cloud-python/blob/75dc3ecefb0ab093f812a81c0c2fccf42407ad10/pubsub/google/cloud/pubsub_v1/subscriber/_protocol/streaming_pull_manager.py#L252-L266).